### DISCUSSION TITLE
Aliasing a field not working

### DISCUSSION THREAD

**Discussion by chirdeeptomar** [mirumee/ariadne#892] [General]
I have a top level field on the Query object called topMetric (which is a generic method) to return top_book, top_authors etc, takes an Emum to distinguish what to return. 

As the field is named as topMetric in the schema, in the output I get "topMetric" under "data" node. I would like to rename that in the resolver based on the type of metric requested from the user. 

user = ObjectType("Query")
user.set_alias("topMetric", "topAuthors")

Above did not work for me. Appreciate any help.

**Comment by rafalp** [comment]
https://ariadnegraphql.org/docs/api-reference.html#set_alias

> `ObjectType.set_alias(name, to)`
> Makes a field `name` defined in the schema resolve to property `to` of an object.

What you want to do is:

```python
user = ObjectType("Query")
user.set_alias("topMetric", "top_metric")
user.set_alias("topAuthors", "top_authors")
```

**Reply by chirdeeptomar** [reply]
Thanks @rafalp, thats what I am doing inside the top_metric resolver. 

`   user = ObjectType("Query")
    user.set_alias("topMetric", "top_metric")`

I am currently just sending the response from the service back straight away. Does my response need to wrapped up in some response object with user explicitly set on that object?

**Reply by rafalp** [reply]
What do you mean by "inside the resolver"? `set_alias` is supposed to be called before the `make_executable_schema`.

**Reply by chirdeeptomar** [reply]
aah right, so I can not dynamically set the alias?

**Reply by rafalp** [reply]
`set_alias` is a shortcut that defines resolver function for given field-attribute name pair and and stores this on `ObjectType` for binding to schema by `make_executable_schema`. If you need to resolve field's value dynamically, you need to do it yourself in resolver function.
